be/src/connector/connector.h

```diff
@@ -180,17 +182,26 @@ class Connector {
     static const std::string FILE;
     static const std::string LAKE;
     static const std::string BINLOG;
+    static const std::string ICEBERG;
 
     virtual ~Connector() = default;
     // First version we use TPlanNode to construct data source provider.
     // Later version we could use user-defined data.
 
     virtual DataSourceProviderPtr create_data_source_provider(ConnectorScanNode* scan_node,
-                                                              const TPlanNode& plan_node) const = 0;
+                                                              const TPlanNode& plan_node) const {
+        CHECK(false) << connector_type() << " connector does not implement chunk source yet";
```

Comment:
Don't use CHECK. It will crash the process

Reply:
The base implementation should never be called. So fail fast here.